modules/entity/spec/sorted_state_adapter.spec.ts

```diff
@@ -407,4 +407,36 @@ describe('Sorted State Adapter', () => {
       },
     });
   });
+
+  it('should let you add one entity to the state with setOne()', () => {
+    const withOneEntity = adapter.setOne(TheGreatGatsby, state);
+    expect(withOneEntity).toEqual({
+      ids: [TheGreatGatsby.id],
+      entities: {
+        [TheGreatGatsby.id]: TheGreatGatsby,
+      },
+    });
+  });
+
+  it('should let you replace an entity in the state with setOne()', () => {
+    const withOne = adapter.addOne(
+      TheGreatGatsby,
+      adapter.addOne(AnimalFarm, adapter.addOne(AClockworkOrange, state))
+    );
+    const updatedBook = {
+      id: TheGreatGatsby.id,
+      title: 'A New Hope',
+      description: undefined,
```

Comment:
Could you please remove the description altogether? With `description: undefined` even `upsertOne` would behave the same way.

Reply:
I can remove it, that is not a problem. But that is the only way to test if 'setOne' actually behaves in a way different from 'upsertOne'. I mean, without this 'undefined' the test does not make much sense. Does it?

Reply:
It actually does make more sense without that property. You can make a comment to be very explicit that we are updating the the entity that doesn't have that property and `toEqual` does the deep equality check.
With the test as-is even `adapter.upsertOne(updatedBook, withOne)` would pass.

I see that this PR is merge, I'd suggest to have a follow-up PR to adjust this test behavoir.

Reply:
Okay, we might have a misunderstanding here... Do you think we need to:
1) remove line 429;
or
2) remove 'description' property from the Book interface altogether?

Reply:
remove the line from 429 and replace with `/* description property is not provided */`